src/main/java/org/reaktivity/nukleus/kafka/internal/stream/KafkaClientProduceFactory.java

```diff
@@ -1506,13 +1528,18 @@ private void doEncodeRecordCont(
         {
             if (value != null)
             {
+                final int length = value.sizeof();
+
+                if (encodeSlotLimit  + length + encodeableRecordHeadersBytes + KAFKA_RECORD_FRAMING >= encodePool.slotCapacity())
```

Comment:
Doesn't `encodeSlotOffset` also factor into this equation?
Suggest ordering the parts in the same way they layout during encoding, such as
```
if (KAFKA_RECORD_FRAMING + (encodeSlotLimit - encodeSlotOffset) + length + encodeableRecordHeadersBytes >= encodePool.slotCapacity())
```

Reply:
encodeSlotLimit should cover it right? At least this is what I understood reading a similar code.

Reply:
Yes, I think using `encodeSlotLimit` is better than `encodeSlotLimit - encodeSlotOffset`, since we are trying to determine when we would hit the limit of the buffer slot, not just the total size of encodable data in the buffer.